### PR TITLE
feat: k8s event listener - RK-19159 

### PR DESCRIPTION
**Pull request by qadiludmer**
Signed-off-by: Adi Ludmer <Adi.Ludmer@qedma.com>

### Description

Please provide a brief description of the changes made in this pull request.

### Related Issue(s)

If this pull request addresses or relates to any open issues, please mention them here using the syntax `Fixes #<issue_number>` or `Resolves #<issue_number>`.

### Checklist

Before submitting this pull request, please ensure that you have completed the following tasks:

- [ ] Reviewed the [Contributing Guidelines](../docs/CONTRIBUTING.md) for the Piper project.
- [ ] Ensured that your changes follow the [coding guidelines and style](../docs/CONTRIBUTING.md#coding-guidelines) of the project.
- [ ] Run the tests locally and made sure they pass.
- [ ] Updated the relevant documentation, if applicable.

### Testing Instructions

Please provide clear instructions on how to test and verify the changes made in this pull request.

### Additional Information

Add any additional information or context that would be helpful in understanding and reviewing this pull request.

### CODE REVIEW

File: pkg/listener/argo_events_broker.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+func (a *ArgoEventsBroker) Start() {
+	// get pod's service account credentials
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Create a new Kubernetes clientset
+	clientset, err := kubernetes.NewForConfig(config)
```
Comment (gosharo) [inline pkg/listener/argo_events_broker.go:32]: we can use our existing client under globalClients.Workflows it has a k8s client

File: pkg/listener/argo_events_broker.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Create a new watcher for Pods in the specified namespace
+	watcher := cache.NewListWatchFromClient(clientset.CoreV1().RESTClient(), "pods", corev1.NamespaceDefault, nil)
```
Comment (gosharo) [inline pkg/listener/argo_events_broker.go:38]: we are looking for Workflows

File: pkg/listener/argo_events_broker.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+	// Start watching for events
+	_, controller := cache.NewInformer(
+		watcher,
+		&corev1.Pod{},
```
Comment (gosharo) [inline pkg/listener/argo_events_broker.go:43]: v1alpha1.Workflow{}

File: pkg/listener/event_broker_example.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package listener
+
+type EventBrokerExample struct {
```
Comment (gosharo) [inline pkg/listener/event_broker_example.go:3]: why do we need an example ?

**Review by gosharo — CHANGES_REQUESTED** [review]

File: pkg/listener/event_broker_example.go (hunk 4 above)
Comment (qadiludmer) [reply pkg/listener/event_broker_example.go]: The example became the real deal 😆

**Pull request closed without merging**